src/main.rs

```diff
@@ -32,78 +57,93 @@ fn main() -> Result<(), Box<dyn Error>> {
+                        KeyCode::Char(' ') | KeyCode::Enter => {
+                            if menu.selection == 0 {
+                                in_menu = false;
+                            } else {
+                                break 'gameloop;
+                            }
                         }
+                        _ => {}
                     }
-                    KeyCode::Esc | KeyCode::Char('q') => {
-                        audio.play("lose");
-                        break 'gameloop;
+                }
+            }
+            menu.draw(&mut curr_frame);
+        } else {
+            // Input handlers for the game
+            while event::poll(Duration::default())? {
+                if let Event::Key(key_event) = event::read()? {
+                    match key_event.code {
+                        KeyCode::Left => player.move_left(),
+                        KeyCode::Right => player.move_right(),
+                        KeyCode::Char(' ') | KeyCode::Enter => {
+                            if player.shoot() {
+                                audio.play("pew");
+                            }
+                        }
+                        KeyCode::Esc | KeyCode::Char('q') => {
+                            audio.play("lose");
+                            in_menu = true;
+                            reset_game(&mut in_menu, &mut player, &mut invaders);
+                        }
+                        _ => {}
                     }
-                    _ => {}
                 }
             }
-        }
 
-        // Updates
-        player.update(delta);
-        if invaders.update(delta) {
-            audio.play("move");
-        }
-        let hits: u16 = player.detect_hits(&mut invaders);
-        if hits > 0 {
-            audio.play("explode");
-            score.add_points(hits);
+            // Updates
+            player.update(delta);
+            if invaders.update(delta) {
+                audio.play("move");
+            }
+            let hits: u16 = player.detect_hits(&mut invaders);
+            if hits > 0 {
+                audio.play("explode");
+                score.add_points(hits);
+            }
+            // Draw & render
+            player.draw(&mut curr_frame);
+            invaders.draw(&mut curr_frame);
+            score.draw(&mut curr_frame);
         }
 
-        // Draw & render
-        let drawables: Vec<&dyn Drawable> = vec![&player, &invaders, &score];
-        for drawable in drawables {
-            drawable.draw(&mut curr_frame);
-        }
         let _ = render_tx.send(curr_frame);
         thread::sleep(Duration::from_millis(1));
 
-        // Win or lose?
-        if invaders.all_killed() {
-            audio.play("win");
-            break 'gameloop;
-        }
-        if invaders.reached_bottom() {
-            audio.play("lose");
-            break 'gameloop;
+        // Final checks if we are in game
+        if !in_menu {
```

Comment:
```suggestion
```

With my earlier suggestion of adding a `continue` in the menu handling, we will no longer reach this code while in a menu.